app/models/deadline_type.rb

```diff
@@ -0,0 +1,302 @@
+# frozen_string_literal: true
+
+# DeadlineType serves as the canonical source of truth for all deadline categories.
+# It replaces hard-coded deadline_type_id comparisons with semantic helper methods.
+class DeadlineType < ApplicationRecord
+  # Constants for deadline type IDs (for backward compatibility)
+  SUBMISSION = 1
+  REVIEW = 2
+  TEAMMATE_REVIEW = 3
+  METAREVIEW = 5
+  DROP_TOPIC = 6
+  SIGNUP = 7
+  TEAM_FORMATION = 8
+  QUIZ = 11
+
+  validates :name, presence: true, uniqueness: true
+  validates :description, presence: true
+
+  has_many :due_dates, foreign_key: :deadline_type_id, dependent: :restrict_with_exception
+
+  # Scopes for categorizing deadline types
+  scope :submission_types, -> { where(name: ['submission']) }
+  scope :review_types, -> { where(name: ['review', 'metareview', 'teammate_review']) }
+  scope :quiz_types, -> { where(name: ['quiz']) }
+  scope :administrative_types, -> { where(name: ['drop_topic', 'signup', 'team_formation']) }
+
+  # Class methods for finding deadline types
```

Comment:
How would you envision these methods being used?  If it's not clear, they should not be included.